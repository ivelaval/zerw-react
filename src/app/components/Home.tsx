import React, { FC } from 'react';

interface IHomeProps {
  userName: string;
  lang: string;
}

export const Home: FC<IHomeProps> = ({ userName, lang }): JSX.Element => {
  return (
    <h1>
      Hi {userName} would you like {lang}?
    </h1>
  );
}
