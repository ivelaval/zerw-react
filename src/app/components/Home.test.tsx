import * as React from 'react';
import { shallow } from 'enzyme';
import { Home } from './Home';

test('Home component after render', () => {
  const home = shallow(<Home userName="Ivan" lang="react" />);

  // Interaction demo
  expect(home.text()).toEqual('Hi Ivan would you like react?');

  // Snapshot demo
  expect(home).toMatchSnapshot();
});
