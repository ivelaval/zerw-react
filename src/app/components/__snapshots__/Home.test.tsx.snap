// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<h1>
  Hi 
  Ivan
   would you like 
  react
  ?
</h1>
`;

exports[`Home component after render 1`] = `
<h1>
  Hi 
  Ivan
   would you like 
  react
  ?
</h1>
`;
